backend/FwHeadless/Program.cs

```diff
@@ -113,6 +129,29 @@ static async Task<Results<Ok<SyncResult>, NotFound, ProblemHttpResult>> ExecuteM
     return TypedResults.Ok(result);
 }
 
+static async Task<Results<Ok<ProjectSyncStatus>, NotFound>> GetMergeStatus(
+    ProjectContext projectContext,
+    ProjectLookupService projectLookupService,
+    ProjectSyncStatusService syncStatusService,
+    IServiceProvider services,
+    LexBoxDbContext lexBoxDb,
+    Guid projectId)
+{
+    var status = syncStatusService.SyncStatus(projectId);
```

Comment:
let's move most of the code in this method inside the `SyncStatus` method in the service, otherwise it's not clear what the difference is between this status and the status we are returning if this is null. Or maybe this should be Job status or something like that considering this indicates if there's an active job or not, then this should return something like `SyncJobStatus.Running` or `SyncJobStatus.NotRunning` and the rest of the code in this method determines the sync status if the job is not running

Reply:
Renamed to SyncJobStatusService in commit 0ce3d271, because that seemed like the cleanest separation of concerns. That service is a singleton only concerned with keeping a ConcurrentDictionary to provide fast, quick lookup of whether a sync is running. The commit-counting logic feels like it belongs elsewhere, so I took the renaming approach.